contracts/WETH10.sol

```diff
@@ -39,29 +40,43 @@ contract WETH10 is IWETH10 {
     /// @dev Current amount of flash-minted WETH10 token.
     uint256 public override flashMinted;
     
+    constructor() {
+        uint256 chainId;
+        assembly {chainId := chainid()}
+        DOMAIN_SEPARATOR = keccak256(
```

Comment:
hardcoding in the constructor a domain separator that depends on the chainId will cause issues when a splitting hard fork happens.
message signed for the original chainId will be replayable on both fork, and user of the the side that use a different chainId will have issues with wallet preventing them to use a different chainId.

Reply:
thanks for flagging this issue @wighawag -- that's a good point. I suppose we could do a check on chainID for each use of `DOMAIN_VARIABLE` like the OZ ref for EIP712 has.... https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/draft-EIP712.sol

otherwise, this seems like an edge case that might introduce some gas costs?  (noting that DAI and USDC appear to face similar issue related to forking)